### PR TITLE
[AJ-1525] Pass Cost Saving Params to LZ when creating Billing Project

### PR DESCRIPTION
**Pull request by ashanhol**
Ticket: https://broadworkbench.atlassian.net/browse/AJ-1525

This PR adds an extra optional argument when creating a billing project that enables cost saving parameters in Landing Zone (specifically, Vertical Pod Autoscaling and Spot Nodes). 
Note: This is a quick PR that unblocks our ability to create workspaces with these features. If we were ever to enable this in the UI we would want to reorchestrate to allow us to pass custom parameters to LZ, which would allow us to enable any number of features on LZ in the future. 

Orchestration's API update: https://github.com/broadinstitute/firecloud-orchestration/pull/1266

---

**PR checklist**

- [x] Include the JIRA issue number in the PR description and title
- [x] Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [x] Get two thumbsworth of PR review
- [x] Verify all tests go green, including CI tests
- [x] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala
Comment (yuliadub) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala:298]: not sure if we want to change this?

Comment (ashanhol) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala:298]: we do, we added a param
so instead of 7 params theres 8

File: core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala
Comment (yuliadub) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala]: it seems like that if the costSaving bool is included but is set to false, this code will still use the cost savings params that are hardcoded to true, or am I reading this wrong?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala
Comment (yuliadub) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserAuth.scala:298]: ah, got it :) tricky scala!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala
Comment (ashanhol) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala]: the full line is 
`val costSavingParams = if (createProjectRequest.costSavings.getOrElse(false)) config.azureConfig.get.costSavingLandingZoneParameters else Map.empty`
so there's an else that sets it to empty like the `maybeAttach` val above it

Comment (davidangb) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala]: Scala syntax goodness: using `contains` might be easier to read because it mentions the word "true":
```scala
if (createProjectRequest.costSavings.contains(true))
```
no functional difference, and this is just style, so your call to change or not

File: core/src/main/scala/org/broadinstitute/dsde/rawls/config/MultiCloudWorkspaceConfig.scala
Comment (davidangb) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/config/MultiCloudWorkspaceConfig.scala]: should these be `String`s? It looks like in rawls.conf the values are all `Boolean`s, though in the test-only reference.conf you have some Strings in there. If you think there's a future use case where we need to pass strings down to WSM/LZ or to Azure for these config settings, leaving this as `String` is fine.

File: local-dev/templates/rawls.conf.ctmpl
Comment (davidangb) [inline local-dev/templates/rawls.conf.ctmpl]: If you store `costSavingLandingZoneParameters` in rawls.conf, you'll need to make a corresponding change to the config in terra-helmfile:
https://github.com/broadinstitute/terra-helmfile/blob/a024f3908c519139ee55dbc6695f62522f53af3c/charts/rawls/templates/config/_conf.tpl#L430. The `local-dev/templates/rawls.conf.ctmpl` file is only used when running Rawls locally.

Alternately, you could put this stanza in [core/src/main/resources/reference.conf](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/resources/reference.conf) and leave it out of rawls.conf altogether.

**reference.conf**: the default values for any setting. Requires a full rebuild/redeploy of Rawls to change, since it's right in the codebase. But, these values are designed to be overridden by values in rawls.conf, so you can effectively change them by overriding. 

**rawls.conf**: the environment-specific values for any setting. Can be changed in k8s "live", with a simple rolling restart, no rebuild or redeploy needed. 

If the `costSavingLandingZoneParameters` config stanza doesn't change between environments (dev vs. prod), and doesn't change frequently, it's a pretty good candidate to store in reference.conf. 

See https://github.com/lightbend/config?tab=readme-ov-file#standard-behavior for understanding how the Typesafe Config library layers these config files. When reading that page, substitute "rawls.conf" anywhere you see "application.conf".

**Review by davidangb — COMMENTED** [review]
fairly minor comments inline, thank you!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala
Comment (yuliadub) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala]: I see, the value get set to false if we cant get it, which means the if will not evaluate to true. I was confusing the getorElse and if/else together. thanks for clearing it up :)

File: core/src/main/scala/org/broadinstitute/dsde/rawls/config/MultiCloudWorkspaceConfig.scala
Comment (ashanhol) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/config/MultiCloudWorkspaceConfig.scala]: This was a mistake, we will be doing future LZ params which might make more sense as strings but not under costSavingLandingZoneParameters. I'll fix it.

Comment (ashanhol) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/config/MultiCloudWorkspaceConfig.scala]: Oh no wait I'm wrong, the expected params for all LZ params are string so we should match

**Review by yuliadub — APPROVED** [review]
I'd suggest we test this in a BEE to make sure everything looks good before merging but LGTM

**Comment by ashanhol** [comment]
jenkins retest

**Comment by ashanhol** [comment]
jenkins retest

**Review by jladieu — APPROVED** [review]

**Comment by ashanhol** [comment]
Note: Billing project deploy from UI works in BEE. Have not been able to test calling the POST billing create api- seems permissions restricted

File: core/src/main/resources/reference.conf
Comment (aherbst-broad) [inline core/src/main/resources/reference.conf:151]: My understanding was that the cost savings VPA + spot nodes stuff was going to be opt-in, is that the case? If so these should parameters should default to false.

Comment (ashanhol) [reply core/src/main/resources/reference.conf:151]: yes thank you, changed. i had modified it in a couple places and forgot to change this back.

Comment (ashanhol) [reply core/src/main/resources/reference.conf:151]: Oh jeez vacation brain, nvm i actually have to reverse this. 
These actually are true by default. The check that enables and unenables them is actually in BpmBillingProjectLifecycle. It checks the user set flag and either pulls these true parameters, or doesn't even use them if the cost savings flag is false.

**Pull request merged**
